Plugins/NativeXr/Source/NativeXr.cpp

```diff
@@ -1706,7 +1707,7 @@ namespace Babylon
             JsRuntimeScheduler m_runtimeScheduler;
             uint32_t m_timestamp{0};
 
-            std::vector<std::pair<const std::string, Napi::FunctionReference>> m_eventNamesAndCallbacks{};
+            std::vector<std::pair<std::string, std::shared_ptr<Napi::FunctionReference>>> m_eventNamesAndCallbacks{};
```

Comment:
why do you need to use shared_ptr?

Reply:
i think i was hitting issues with the const std::string, retesting this again it doesn't seem needed. It has been reverted.